nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+
+    private Map<String, String> buildNewAttributes(FlowFile file, String atrList,
+                                                   EncryptContent.Encryptor encryptor,
+                                                   boolean isToBeEncrypted) throws Exception {
+
+        Map<String, String> oldAttributes = file.getAttributes();
+        Map<String, String> atrToWrite = new HashMap<>();
+        final String filenameAttr = CoreAttributes.FILENAME.key();
+        final String uuidAttr = CoreAttributes.UUID.key();
+
+        if (!StringUtils.isEmpty(atrList)) {
+            //TODO: check for spaces also
+            // Traverse comma separated list if provided
+            String[] attrs = atrList.split(",");
+            Set<String> atrSet = new HashSet<>(Arrays.asList(attrs));
+            for (String atr : atrSet) {
+                if (oldAttributes.containsKey(atr) && !atr.equals(filenameAttr) && !atr.equals(uuidAttr)) {
+                    String atrValue = oldAttributes.get(atr);
+                    String newAtrVal = (isToBeEncrypted) ? EncryptString.performEncryption(atrValue, encryptor) : EncryptString.performDecryption(atrValue, encryptor);
+                    atrToWrite.put(atr, newAtrVal);
+                }
+            }
+        } else {
+            //TODO: filename encryption check only for PGP algos.
+            // encrypt all attributes except filename and uuid.
+            for (String atr : oldAttributes.keySet()) {
+                if (!atr.equals(filenameAttr) && !atr.equals(uuidAttr)) {
+                    String atrValue = oldAttributes.get(atr);
+                    String newAtrVal = (isToBeEncrypted) ? EncryptString.performEncryption(atrValue, encryptor) : EncryptString.performDecryption(atrValue, encryptor);
+                    atrToWrite.put(atr, newAtrVal);
+                }
+            }
+        }
+        return atrToWrite;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        final String atrList = context.getProperty(ATTRIBUTES_TO_ENCRYPT).getValue();
+        final String method = context.getProperty(ENCRYPTION_ALGORITHM).getValue();
+        final EncryptionMethod encryptionMethod = EncryptionMethod.valueOf(method);
+        final String providerName = encryptionMethod.getProvider();
+        final String algorithm = encryptionMethod.getAlgorithm();
+        final String password = context.getProperty(PASSWORD).getValue();
+        final KeyDerivationFunction kdf = KeyDerivationFunction.valueOf(context.getProperty(KEY_DERIVATION_FUNCTION).getValue());
+        final boolean encrypt = context.getProperty(MODE).getValue().equalsIgnoreCase(ENCRYPT_MODE);
+
+        EncryptContent.Encryptor encryptor;
+        Map<String, String> newAtrList;
+
+        try {
```

Comment:
It could be called `EncryptionProcessorUtils` or something similar. Yes, it needs separate unit tests, but this should be easy as there is existing application code which exercises the methods, and the methods should be specific and contained enough to quickly capture the positive/negative cases.